src/lib.rs

```diff
@@ -311,3 +346,82 @@ impl Write for Device {
         Ok(())
     }
 }
+
+impl Device {
+    pub fn set_mpsse_clock(&mut self, freq: u32) -> std::result::Result<(), io::Error> {
+        const MAX: u32 = 30_000_000;
+        const MIN: u32 = 92;
+
+        assert!(
+            freq >= MIN,
+            "frequency of {} exceeds minimum of {}",
+            freq,
+            MIN
+        );
+        assert!(
+            freq <= MAX,
+            "frequency of {} exceeds maximum of {}",
+            freq,
+            MAX
+        );
+
+        let (divisor, clkdiv) = if freq <= 6_000_000 {
+            (6_000_000 / freq - 1, Some(true))
+        } else {
+            (30_000_000 / freq - 1, Some(false))
+        };
+
+        self.write_all(MpsseCmdBuilder::new().set_clock(divisor, clkdiv).as_slice())?;
+
+        Ok(())
+    }
+}
+
+impl MpsseCmdExecutor for Device {
+    type Error = io::Error;
```

Comment:
This is actually an interesting question. `write_all` and `read_exact` only use `io::Error` to conform to standard library's `Read` and `Write` traits. I'm not sure which error type is most appropriate here, so I'm fine with landing with `io::Error`, but this question may be interesting from the consumer point of view.

Reply:
Well, anything can be supported by `MpsseCmdExecutor` users including `ftdi-embedded-hal`. For instance, custom or aggregated error type. For the sake of simplicity I suggest to use the error type returned by `io::Read` and `io::Write` trait implementations in `ftdi-rs`.

Reply:
Okay, leaving this as is.